woocommerce-myparcel.php

```diff
@@ -120,16 +121,24 @@ private function getVersion(): string
          */
         private function setupWebhooks(): void
         {
-            $changeOrderStatusAfter = WCMP_Export_Consignments::getSetting(
-                WCMYPA_Settings::SETTING_CHANGE_ORDER_STATUS_AFTER
-            );
-            $exportMode             = WCMP_Export_Consignments::getSetting(WCMYPA_Settings::SETTING_EXPORT_MODE);
+            $currentApiKey = $this->setting_collection->getByName(WCMYPA_Settings::SETTING_API_KEY);
+            $validApiKey   = get_option('woocommerce_myparcel_valid_api_key');
 
-            if (WCMP_Settings_Data::CHANGE_STATUS_AFTER_PRINTING === $changeOrderStatusAfter && WCMP_Settings_Data::EXPORT_MODE_PPS === $exportMode) {
-                (new OrderStatusWebhook())->register();
-            }
+            if ($validApiKey !== $currentApiKey) {
```

Comment:
Ik vind $validApiKey een beetje misleidend als het klopt wat ik in de code lees, want dan wordt niet gecheckt of de api key valid is. En wat is er met de apikeysettingslistener

Reply:
Het lijkt erop alsof je hier een boolean gaat checken tegen een string, dit zal toch altijd false zijn?